nav2_mppi_controller/include/nav2_mppi_controller/tools/utils.hpp

```diff
@@ -276,8 +265,8 @@ inline bool withinPositionGoalTolerance(
 template<typename T>
 auto normalize_angles(const T & angles)
```

Comment:
This function I noted some energy in playing with exactly how to formulate it since the flamegraphs showed me that this was where non-trivial amounts of time were spent. Did you experiment with a few ways and see that this was indeed the fastest? Is this a chokepoint in gdb?

Reply:
In general, have you looked at some flame graphs / gdb output to see where the main bottlenecks are that might be worth revisiting now with Eigen

Reply:
I have not profiled them using a profiler, but I have manually compared the timings of many Eigen methods to see which one is faster. However, there are some places where redundant copies might be happening. RollColumns method seems to be one.

Reply:
@Ayush1285 just to sanity check here again - did you test this function though and the method used is the fastest?

Reply:
Yes, I've compared unaryExpr and manual for looping, and there was no difference for an array size of 10 million.